flink-streaming-java/src/main/java/org/apache/flink/streaming/api/operators/source/WatermarkToDataOutput.java

```diff
@@ -83,4 +80,25 @@ public void markIdle() {
             throw new ExceptionInChainedOperatorException(e);
         }
     }
+
+    @Override
+    public void markActive() {
+        try {
+            markActiveInternally();
+        } catch (ExceptionInChainedOperatorException e) {
+            throw e;
+        } catch (Exception e) {
+            throw new ExceptionInChainedOperatorException(e);
+        }
+    }
```

Comment:
Why do we handle the exceptions differently than before?

Reply:
This method did not exist before; I copied the handling from #markIdle and #emitWatermark without thinking too much tbh.